internal/versionsapi/cli/rm.go

```diff
@@ -0,0 +1,753 @@
+/*
+Copyright (c) Edgeless Systems GmbH
+
+SPDX-License-Identifier: AGPL-3.0-only
+*/
+
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"regexp"
+	"strings"
+	"time"
+
+	compute "cloud.google.com/go/compute/apiv1"
+	"cloud.google.com/go/compute/apiv1/computepb"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	armcomputev4 "github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/compute/armcompute/v4"
+	awsconfig "github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+	"github.com/aws/smithy-go"
+	"github.com/edgelesssys/constellation/v2/internal/logger"
+	"github.com/edgelesssys/constellation/v2/internal/versionsapi"
+	verclient "github.com/edgelesssys/constellation/v2/internal/versionsapi/client"
+	gaxv2 "github.com/googleapis/gax-go/v2"
+	"github.com/spf13/cobra"
+	"go.uber.org/multierr"
+	"go.uber.org/zap/zapcore"
+)
+
+func newRmCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "rm",
+		Short: "Remove an object",
+		Long:  "Remove an object from the versions API.",
+		RunE:  runRm,
+		Args:  cobra.ExactArgs(0),
+	}
+
+	cmd.Flags().String("ref", "", "Ref to delete from.")
+	cmd.Flags().String("stream", "", "Stream to delete from.")
+	cmd.Flags().String("version", "", "Version to delete. The versioned objects are deleted.")
+	cmd.Flags().String("version-path", "", "Short path of a single version to delete. The versioned objects are deleted.")
+	cmd.Flags().Bool("all", false, "Delete the entire ref. All versions and versioned objects are deleted.")
```

Comment:
Should we add a safeguard here as to not accidentally delete all images in important refs sch as `-`? 
An additional user prompt which echos the image to be deleted could also help prevent such errors.

Reply:
:D  I had the same idea, implemented in [ac8f420](https://github.com/edgelesssys/constellation/pull/840/commits/ac8f420d4906cb4b23262251bf6e3b8587113ed7). Maybe I should also add `main`. Not sure about the prompting, the CLI is heavily used in CI and I would prefer to not add more complexity as necessary.

Reply:
I'm thinking about adding an env check so that removing is only allowed in CI, most devs won't have permissions to delete on GCP anyway.

Reply:
@malt3 is against an env check. Maybe a disclaimer will do...

Reply:
I've added a disclaimer, see [db4e826](https://github.com/edgelesssys/constellation/pull/840/commits/db4e8265a806f5839eac122e16e46de6b91aff6d) and [c8be479](https://github.com/edgelesssys/constellation/pull/840/commits/c8be4797e8c3c4c71973ad877c98456caa79f437)